specs/models/Step.v1.json

```diff
@@ -0,0 +1,181 @@
+{
+  "title": "Step.v1",
+  "type": "object",
+  "properties": {
+    "options": {
+      "type": "object",
+      "properties": {
+        "sourceQuery": {
+          "type": "string",
+          "description": "Defines the items to be processed by the step; must be a cts.query or cts.uris statement if sourceQueryIsScript is false"
+        },
+        "sourceQueryIsScript": {
+          "type": "boolean",
+          "description": "Added in 5.3.0; if true, then sourceQuery can be any JavaScript statement that can be passed into xdmp.eval"
+        },
+        "constrainSourceQueryToJob": {
+          "type": "boolean",
+          "description": "If true, the query is applied to the documents that were created or modified in the same job that executes the step"
+        },
+        "provenanceGranularityLevel": {
+          "type": "string",
+          "description": "The granularity of the provenance tracking information: coarse (default) to store document-level provenance information only, fine to store document-level and property-level provenance information, or off to disable provenance tracking in future job runs. Applies only to mapping, matching, merging, mastering, and custom steps.",
+          "enum": [
+            "off",
+            "coarse",
+            "fine"
+          ]
+        },
+        "stepUpdate": {
+          "type": "boolean",
+          "description": "If true, custom modules can make changes directly to records in the database"
+        },
+        "acceptsBatch": {
+          "type": "boolean",
+          "description": "If true, the step module is invoked once with all records in the batch passed to it"
+        },
+        "collections": {
```

Comment:
Taking a page out of mlcp's book here - how about "outputCollections" to make it clear that these are applied to documents outputted by the step? Likewise for "outputPermissions", and I'd do "outputAdditionalCollections" as well. I remember being confused at first about what the scope of "collections" is - does that affect the query for documents to process, or are they applied to documents that will be written, or is it something else? "outputCollections" works well for mlcp, I think we should copy it here.

(Note that the function that adapts the new Step schema to what the existing Flow schema expects would then rename this to "collections").

Reply:
I've updated this to use "outputCollections" and "defaultOutputCollections".